httpie/client.py

```diff
@@ -190,10 +191,37 @@ def finalize_headers(headers: RequestHeadersDict) -> RequestHeadersDict:
             if isinstance(value, str):
                 # See <https://github.com/httpie/httpie/issues/212>
                 value = value.encode()
-        final_headers[name] = value
+        final_headers.add(name, value)
     return final_headers
 
 
+def apply_missing_repeated_headers(
+    prepared_request: requests.PreparedRequest,
+    original_headers: RequestHeadersDict
+) -> None:
+    """Update the given `prepared_request`'s headers with the original
+    ones. This allows the requests to be prepared as usual, and then later
+    merged with headers that are specified multiple times."""
+
+    new_headers = RequestHeadersDict(prepared_request.headers)
```

Comment:
Couldn’t we just `.copy()` the headers?

Reply:
`prepared_request.headers` is a regular dict, we are casting it into a `RequestHeadersDict`.